widget/list.go

```diff
@@ -116,17 +123,23 @@ func (l *listRenderer) Layout(size fyne.Size) {
 
 	// Relayout What Is Visible - no scroll change - initial layout or possibly from a resize.
 	l.visibleItemCount = int(math.Ceil(float64(l.scroller.size.Height) / float64(l.list.itemMin.Height)))
+	if len(l.children) > l.list.Length() {
+		for i := len(l.children); i >= l.list.Length(); i-- {
+			l.itemPool.Release(l.children[i-1])
+		}
+		l.children = l.children[:l.list.Length()-1]
+	}
 	if len(l.children) > l.visibleItemCount {
 		for i := len(l.children); i > l.visibleItemCount; i-- {
-			l.itemPool.Release(l.children[len(l.children)-1])
-			l.children = l.children[:len(l.children)-1]
+			l.itemPool.Release(l.children[i-1])
 		}
+		l.children = l.children[:l.visibleItemCount]
 	}
 	if l.visibleItemCount > 0 && l.list.Length() < l.visibleItemCount && len(l.children) > l.visibleItemCount-1 {
```

Comment:
With the fix in place above it looks like this code will no longer be executed - it iterates through the difference of visibleItemCount and list.Length(), but between the code loops above are you not ensuring that those items are already gone?